modules/aws-kernel/src-native/smithy4s/aws/CryptoPlatformCompat.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.aws.kernel
+
+import scala.scalanative.unsafe
+import scala.scalanative.unsafe._
+import scala.scalanative.unsigned._
+
+trait CryptoPlatformCompat {
+  type Binary = Array[Byte]
+
+  def binaryFromString(str: String): Binary =
+    str.getBytes("UTF-8")
+
+  def toHexString(binary: Binary): String =
+    binary.map("%02x".format(_)).mkString
+
+  def sha256HexDigest(message: Array[Byte]): String =
+    unsafe.Zone { implicit z =>
+      val outLength: Ptr[CUnsignedInt] = unsafe.alloc[CUnsignedInt]()
+      val digest: Ptr[CUnsignedChar] =
+        unsafe.alloc[CUnsignedChar](CryptoPlatformCompat.EVP_MAX_MD_SIZE)
+      val dataLen = message.length.toLong
+      val cInput: CString = arrayToCString(message)
+
+      val ctx = crypto.EVP_MD_CTX_new()
+      crypto.EVP_DigestInit(ctx, crypto.EVP_sha256())
+      crypto.EVP_DigestUpdate(ctx, cInput, dataLen.toUInt)
+      crypto.EVP_DigestFinal(ctx, digest, outLength)
+
+      val out = Array.fill[Byte]((!outLength).toInt)(0)
+      for (i <- 0 until (!outLength).toInt) out(i) = digest(i.toLong).toByte
+      out.map("%02x".format(_)).mkString
+    }
+
+  def sha256HexDigest(message: String): String = sha256HexDigest(
+    message.getBytes("UTF-8")
+  )
+
+  def hmacSha256(data: String, key: Binary): Binary =
+    unsafe.Zone { implicit z =>
+      val outLength: Ptr[CUnsignedInt] = unsafe.alloc[CUnsignedInt]()
+      val digest: Ptr[CUnsignedChar] =
+        unsafe.alloc[CUnsignedChar](CryptoPlatformCompat.EVP_MAX_MD_SIZE)
+
+      val cKey = arrayToCString(key)
+      val keylen = key.length.toLong
+
+      val cData: CString = toCString(data)
+      val datalen = data.length.toUInt
```

Comment:
this is also different - here and in other places I had to do `.toUInt` to convert to `CSize` - apparently it's a change from SN 0.4 that made the type unsigned: https://github.com/scala-native/scala-native/pull/1949/